tests/test_specdataextractor.py

```diff
@@ -0,0 +1,52 @@
+import json
+import os
+
+import pytest
+
+from infra.system.plugins.specdataextractor.extractor import SpecDataExtractor
+from infra.system.helpers.artefact_schema_validator import ArtefactSchemaValidator
+
+
+CONFIG_FILE_NAME = 'specdataextractor.json'
+
+
+configuration = {}
+
+
+def pytest_generate_tests(metafunc):
+    path = os.path.dirname(os.path.realpath(__file__))
+    configfile = os.path.join(path, CONFIG_FILE_NAME)
+    try:
+        with open(configfile, 'rb') as f:
+            configuration.update(json.load(f))
+    except IOError:
+        raise RuntimeError(
+            'Cannot load configuration from "{}"'.format(configfile))
+    if 'package' in metafunc.fixturenames:
+        metafunc.parametrize('package', configuration['packages'])
+
+
+class TestSpecDataExtractor(object):
+
+    @pytest.fixture(autouse=True)
+    def prepare(self, request, package):
+        specfile = os.path.join(
+            configuration['testdatadir'], package['specfile'])
+        self.input_data = {
+            'product': package['product'],
+            'distribution': package['distribution'],
+            'package': package['name'],
+            'resource': specfile,
+        }
+
+    def test_valid_output(self, package):
+        plugin = SpecDataExtractor()
+        assert plugin.setData(self.input_data)
+        assert plugin.execute()
+        output_data = plugin.getData()
+        validator = ArtefactSchemaValidator('golang-project-info-fedora')
```

Comment:
Can you validate all artefacts in a loop so you don't have to assume ordering of artefacts?

Instead of ``'golang-project-info-fedora'`` could you import artefact constant from ``infra.system.artefacts.artefacts``? The same for the remaining two artefacts?